### PR TITLE
ENH: Add channel parameter to sensirion_ekh4

### PR DESCRIPTION
**Pull request by michaeldaas**
This PR adds a `channel` parameter to the thre `get_` methods of the `sensirion_ekh4` HL. This enables using `FunctionalRegisters` to call each sensor connected to the Sensirion Multiplexer box as independent device.

### CODE REVIEW

File: basil/HL/sensirion_ekh4.py [hunk 1]
```diff
@@ -86,7 +93,11 @@ def get_dew_point(self, min_val=-40, max_val=100):
             else:
                 values.append(self.cal_ret(
                     ret[2 + 2 * j] + ret[2 + 2 * j + 1]))
-        return values
+
+        if channel == 'all':
+            return values
+        elif isinstance(channel, int):
+            return values[channel]
```
Comment (themperek) [inline basil/HL/sensirion_ekh4.py]: Should in return a list even if one channel?

**Comment by themperek** [comment]
Do you really need this? You can just do the same in the user code?

**Comment by laborleben** [comment]
This should do it 👍

**Comment by michaeldaas** [comment]
> Do you really need this? You can just do the same in the user code?

i don't think i can use functional registers in my yaml file to give individual names to the channels without this feature, can i?

File: basil/HL/sensirion_ekh4.py (hunk 1 above)
Comment (michaeldaas) [reply basil/HL/sensirion_ekh4.py]: i don't know what's best here?

File: basil/HL/sensirion_ekh4.py [hunk 2]
```diff
@@ -75,9 +78,13 @@ def get_humidity(self, min_val=0, max_val=100):
             else:
                 values.append(self.cal_ret(
                     ret[2 + 2 * j] + ret[2 + 2 * j + 1]))
-        return values
 
-    def get_dew_point(self, min_val=-40, max_val=100):
+        if channel == 'all':
```
[the remark this answers is not in the document]
Comment (michaeldaas) [reply basil/HL/sensirion_ekh4.py]: done

File: basil/HL/sensirion_ekh4.py [hunk 3]
```diff
@@ -86,7 +91,10 @@ def get_dew_point(self, min_val=-40, max_val=100):
             else:
                 values.append(self.cal_ret(
                     ret[2 + 2 * j] + ret[2 + 2 * j + 1]))
-        return values
+
+        if channelis None:
```
Comment (DavidLP) [inline basil/HL/sensirion_ekh4.py]: Well, broken pychecker?

Comment (michaeldaas) [reply basil/HL/sensirion_ekh4.py]: triggerhappy commit-fingers :see_no_evil:

File: basil/HL/sensirion_ekh4.py (hunk 1 above)
Comment (DavidLP) [reply basil/HL/sensirion_ekh4.py]: Would return number or list. IF you return always list one cannot use return value directly and this feature would be useless, to my mind.

Comment (themperek) [reply basil/HL/sensirion_ekh4.py]: Fine with me anyway if usefull.

**Comment by michaeldaas** [comment]
any more changes necessary?

**Pull request merged**